auth/client-api.md

```diff
@@ -25,7 +25,7 @@ abstract class Client {
   // ---------- Events ----------------------------------------------- //
 
   // subscribe to auth response
-  public abstract on("auth_response", (id: number, cacao: Cacao) => {}): void;
+  public abstract on("auth_response", (id: number, result: Address | Error) => {}): void;
```

Comment:
What is the success type `Address` here?

Reply:
I think it's better to stay with Cacao

Reply:
of Address I thought as of typealias for String.
@Elyniss why you think it is better to stay with cacao?
I think the only info dapp needs is an address if authentication was successful. Additionally cacao issuer is didpkh so it quite inconvenient to derive an address. Plus if app needs some data from request/response it is suppose to be able to query request by id from a client

Reply:
I agree that for use cases 99% only `address` would suffice, but what if dapp would like some data about `nbf` and `exp`?

Reply:
`nbf` and `exp` is set by dapp so dapp is suppose to know it and can querry it by calling:

```
public abstract getRequest(params: { id: number }): Promise<Cacao>;
```

actually I would change the return value as well for something more precise, maybe is cacao valid?

Reply:
Yeah, you're right about `nbf` and `exp`. I agree then only returning address would suffice.

However, I would not change this `public abstract getRequest(params: { id: number }): Promise<Cacao>;`. If we keep it we don't force dapps to store whole cacao as only id would suffice. But I don't have strong opinion on this. I only think that leaving it is a better dev experience.

Reply:
> Yeah, you're right about nbf and exp. I agree then only returning address would suffice.
Now I do not agree with you 🤣  and I think you had a good point....
I suppose dapp is going to construct jwt from cacao properties so it may be useful to have cacao in the event 🤔

Reply:
> However, I would not change this public abstract getRequest(params: { id: number }): Promise<Cacao>;. If we keep it we don't force dapps to store whole cacao as only id would suffice. But I don't have strong opinion on this. I only think that leaving it is a better dev experience.

I think you misunderstood, I meant that cacao man not be enough 🤔 
because what if response was an error or signature validation failed? should we return invalid cacao with this function, without telling is invalid? something else?

Reply:
anyway:
```
  public abstract on("auth_response", (id: number, result: (cacao, string) | Error) => {}): void;
```
where string is an address?
too verbose?

Reply:
I'm still more favouring the whole Cacao.

Reply:
to which `auth_response` ?